src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -145,4 +146,27 @@ trait LibraryServiceSupport extends LazyLogging {
     docs.copy(results = updatedResults)
   }
 
+  // this method will determine if the user is makings a change to discoverableByGroups
+  // if the attribute does not exist on the workspace, it is the same as the empty list
+  def isDiscoverableDifferent(workspaceResponse: WorkspaceResponse, userAttrs: AttributeMap): Boolean = {
+
+    //This converts the Seq of Attribute to a Seq of String (or the empty list) so we can compare the strings,
+    def convert(list: Option[Attribute]): Seq[String] = {
+      list match {
+        case Some(x) if x.isInstanceOf[AttributeValueList] => x.asInstanceOf[AttributeValueList].list.asInstanceOf[Seq[AttributeString]] map { str => str.value }
+        case _ => Seq.empty
+      }
+    }
+    val current = convert(workspaceResponse.workspace.attributes.get(LibraryService.discoverableWSAttribute))
+    val newvals = convert(userAttrs.get(LibraryService.discoverableWSAttribute))
+
+    if (current.isEmpty && newvals.isEmpty)
+      false
+    else if (current.nonEmpty && newvals.nonEmpty) {
+      current.toSet != newvals.toSet
+    } else
+      // one of the values is the empty list and the other is not
+      true
+  }
+
```

Comment:
if you ignored the corner case where the user is changing the order of values within the list - and it's definitely a corner case because the UI doesn't allow lists - this would be as simple as:

```
  def isDiscoverableDifferent(workspaceResponse: WorkspaceResponse, userAttrs: AttributeMap): Boolean = {

    val current = workspaceResponse.workspace.attributes.get(LibraryService.discoverableWSAttribute)
    val newvals = userAttrs.get(LibraryService.discoverableWSAttribute)

    current != newvals
  }
```

is the corner case worth the extra code?

If you _do_ feel adamant about ignoring order, how about simplifying to this:

```
  def isDiscoverableDifferent(workspaceResponse: WorkspaceResponse, userAttrs: AttributeMap): Boolean = {

    def convert(list: Option[Attribute]): Seq[AttributeValue] = {
      list match {
        case Some(x:AttributeValueList) => x.list
        case _ => Seq.empty[AttributeValue]
      }
    }
    val current = convert(workspaceResponse.workspace.attributes.get(LibraryService.discoverableWSAttribute))
    val newvals = convert(userAttrs.get(LibraryService.discoverableWSAttribute))

    current.toSet != newvals.toSet

  }
```